contrib/platform/src/com/sun/jna/platform/WindowUtils.java

```diff
@@ -1297,11 +1297,38 @@ public String getProcessFilePath(final HWND hwnd) {
                 final int length = Psapi.INSTANCE.GetModuleFileNameExW(process,
                                                                        null, filePath, filePath.length);
                 if (length == 0) {
-                    if(Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_INVALID_HANDLE) {
-                        throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
-                    } else {
+                    int lastError = Kernel32.INSTANCE.GetLastError();
+                    if(lastError == WinNT.ERROR_INVALID_HANDLE) {
                         // ignore invalid handles
                         return "";
+                    } else if (lastError == WinNT.ERROR_PARTIAL_COPY) {
+                        // observed on 64 platforms when the JVM running the JNA
+                        // code is 32 bit and the target process is 64 bit.
+                        // Fallback to GetProcessImageFileName. But the returned
+                        // path is in device form, so this needs to be mapped
+                        // back to the drive form
+                        final int length2 = Psapi.INSTANCE.GetProcessImageFileName(process,
+                            filePath, filePath.length);
+                        if (length2 == 0) {
+                            throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+                        }
+                        String processImagePath = Native.toString(filePath);
+                        char[] pathPrefixBuffer = new char[1024];
+                        // Map DOS Drives to drive form paths
+                        for (char i = 'A'; i <= 'Z'; i++) {
```

Comment:
May I suggest rather than iterating every drive letter, you use the FileApi functions ([FindFirstVolume](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findfirstvolumew), [FindNextVolume](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findnextvolumew), and [FindVolumeClose](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findvolumeclose))?  I've implemented this using JNA code [here](https://github.com/oshi/oshi/blob/e76baf5fe21427145012151344b0191e3c00987a/oshi-core/src/main/java/oshi/software/os/windows/WindowsFileSystem.java#L183) although there's extra stuff in there you don't need.

See also sample Windows code [here](https://docs.microsoft.com/en-us/windows/win32/fileio/displaying-volume-paths) referencing those API calls.